File: uncategorized/3005_count_elements_with_maximum_frequency.ts
function maxFrequencyElements(nums: number[]): number {

    let counts = {};
    let mx = 0;
    let ans = 0;

    for(let num of nums){
        counts[num] = (counts[num] || 0) + 1;
        mx = Math.max(mx, counts[num]);
    }

    for(let v of Object.values(counts)){
        if(v === mx){
            ans += v;
        }
    }

    return ans;
};

File: uncategorized/3005_count_elements_with_maximum_frequency.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_frequency_elements(nums: Vec<i32>) -> i32 {

        let mut ans:i32 = 0;
        let mut mx:i32 = 0;
        let mut counts:HashMap<i32, i32> = HashMap::new();

        for num in nums.iter() {
            let count = counts.entry(*num).or_insert(0);
            *count += 1;
            mx = i32::max(mx, *count);
        }

        for (i,v) in counts.iter() {
            if *v == mx {
                ans += v;
            }
        }

        ans

    }
}